io/net/ethtool_test.py

```diff
@@ -128,17 +133,77 @@ def test_ethtool(self):
         '''
         Test the ethtool args provided
         '''
+        global param, default_set
         for state, status in zip(["down", "up"], ["no", "yes"]):
             if not self.interface_state_change(self.iface, state, status):
                 self.fail("interface %s failed" % state)
-            cmd = "ethtool %s %s %s" % (self.args, self.iface, self.elapse)
-            ret = process.run(cmd, shell=True, verbose=True,
-                              ignore_status=True)
-            if ret.exit_status != 0:
-                if "Operation not supported" in ret.stderr_text:
-                    self.log.warn("%s failed" % self.args)
+            value = [self.tx, self.rx, self.other, self.combined]
+            param = ['tx', 'rx', 'other', 'combined']
+            default = []
+            if self.args == "-L":
+                cmd_l = "ethtool %s %s %s" % ("-l", self.iface, self.elapse)
+                output = process.run(cmd_l, shell=True, verbose=True,
+                                     ignore_status=True).stdout_text \
+                                                        .splitlines()[2:6]
+                for i in range(len(output)):
+                    default.append(output[i].split(':')[1])
+                    if 'n/a' in output[i]:
+                        param[i], value[i], default[i] = '', '', ''
+                default_set = default.copy()
+                elements = all([elem == '' for elem in value])
+                if elements:
+                    self.log.warn("Cannot set device channel for null")
                 else:
-                    self.fail("failed")
+                    for i in range(4):
+                        if default[i] != '':
+                            default[i] = ['0', '1', int(default[i])//2,
+                                          default[i], int(default[i])+1]
+                            for j in range(5):
+                                if value[i] != '':
+                                    cmd = "ethtool %s %s %s %s" % (
+                                            self.args, self.iface,
+                                            param[i], default[i][j])
+                                    result = process.run(cmd, shell=True,
+                                                         verbose=True,
+                                                         ignore_status=True)
+                                    if state is 'up':
+                                        ping = self.networkinterface.
+                                        ping_check(self.peer, count=5)
```

Comment:
ah it is confusing can calling function in single line

Reply:
@PraveenPenguin  as we know we cannot enter characters more than 79 to support pycodestyling , i had to split if condition in 3 lines which was making more confusing, thought of making if condition understandable for reader created a variable "ping".
Creating a variable is not increasing number of lines, instead of splitting if condition in 3 lines splitted 'ping' variable in 2 lines and made if condition in 1 line.

Reply:
i say in that case it will be fine as having code wrap in 80 char is good paractice